compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/DomInputStrategy.kt

```diff
@@ -0,0 +1,174 @@
+package androidx.compose.ui.platform
+
+import androidx.compose.ui.text.input.CommitTextCommand
+import androidx.compose.ui.text.input.DeleteSurroundingTextInCodePointsCommand
+import androidx.compose.ui.text.input.EditCommand
+import androidx.compose.ui.text.input.SetComposingTextCommand
+import androidx.compose.ui.text.input.TextFieldValue
+import org.w3c.dom.HTMLTextAreaElement
+import org.w3c.dom.events.CompositionEvent
+import org.w3c.dom.events.Event
+import org.w3c.dom.events.KeyboardEvent
+
+internal interface DomInputStrategy {
+    val htmlInput: HTMLTextAreaElement
+    fun updateState(textFieldValue: TextFieldValue)
+}
+
+internal class CommonDomInputStrategy(
+    override val htmlInput: HTMLTextAreaElement,
+    private val composeSender: ComposeCommandCommunicator,
+) : DomInputStrategy {
+
+    private var lastActualCompositionTimestamp: Int = 0
+    private var editState: EditState = EditState.Default
+
+
+    init {
+        initEvents()
+    }
+
+    override fun updateState(textFieldValue: TextFieldValue) {
+        if (editState != EditState.WaitingComposeActivity) return
+
+        htmlInput.value = textFieldValue.text
+        htmlInput.setSelectionRange(textFieldValue.selection.start, textFieldValue.selection.end)
+
+        editState = EditState.Default
+    }
+
+    private fun initEvents() {
+        var lastKeyboardEventIsDown = false
+
+        htmlInput.addEventListener("blur", {evt ->
+            // both accent dialogue and composition dialogue are lost when we switch windows
+            // but can be restored later on if we are back
+            editState = EditState.Default
+        })
+
+        htmlInput.addEventListener("keydown", {evt ->
+            evt as KeyboardEvent
+            lastKeyboardEventIsDown = evt.key != "Dead" && evt.key != "Unidentified"
+
+            if (editState is EditState.AccentDialogue) {
+                evt.preventDefault()
+                return@addEventListener
+            }
+
+            if (evt.repeat) {
```

Comment:
As far as I know, on Windows a key long press doesn't trigger any AccentDialogue.  
Do we need a condition here?

Reply:
Actually, an OS-based condition is not a good idea. Because it's possible to disable the accents dialog on macOS:

```
defaults write -g ApplePressAndHoldEnabled -bool false
```
https://apple.stackexchange.com/questions/332769/macos-disable-popup-showing-accented-characters-when-holding-down-a-key

Reply:
Looks like to follow the OS behaviour and respect the user-settings, we would need to use the backing text area as a source of truth and "relay" the state of backing text area to Compose TextField.

 https://docs.flutter.dev/cookbook/forms/text-input (there is a demo to play with)

With such approach, it likely means that Compose-defined short-cuts (key mappings) won't  be taken into account. If it become a problem for someone, we can add some flag to disable backing text area.

Reply:
At the same time, the TextField's onValueChange might modify / transofrm the value before setting. In this case the value in backing text area must be updated too.
I'm not sure yet how to achieve this.

Reply:
In google docs I see it's doing something similar:
```
<iframe class="docs-texteventtarget-iframe docs-offscreen-z-index docs-texteventtarget-iframe-negative-top" scrolling="no" src="about:blank" style="contain: strict; top: 0px; left: 0px; position: fixed; transform: matrix(1, 0, 0, 1, 0, -10000); transform-origin: 0px 0px;"></iframe>
```

Reply:
@eymar I'll do everything within my capacities in attempts resolve everything that possible on compose side, exactly (aong other things) because of the onValueChange existence